cmd/vulcan-nuclei/main.go

```diff
@@ -0,0 +1,269 @@
+			if err != nil {
+				logger.Errorf("exec.Command() with args [%s] failed with %s\nCommand output: %s\n", args, err, string(cmdOutput))
+				return errors.New("nuclei command execution failed")
+			}
+			logger.Infof("nuclei command with args [%s] execution completed successfully", args)
+			return nil
+		},
+		retry.Attempts(3),
+		retry.DelayType(retry.RandomDelay),
+		retry.MaxJitter(5*time.Second),
+	)
+	if err != nil {
+		return []byte{}, err
+	}
+	return cmdOutput, nil
+}
+
+func run(ctx context.Context, target, assetType, optJSON string, state checkstate.State) error {
+	// TODO: If options are "malformed" perhaps we should not return error
+	// but only log and error and return.
+	var opt options
+	if optJSON != "" {
+		if err := json.Unmarshal([]byte(optJSON), &opt); err != nil {
+			return err
+		}
+	}
+
+	isReachable := helpers.IsWebAddrsReachable(target)
+	if !isReachable {
+		return checkstate.ErrAssetUnreachable
+	}
+
+	if opt.ForceUpdateTemplates {
+		_, err := runNucleiCmd([]string{"-update-templates"})
+		if err != nil {
+			return errors.New("nuclei update-templates command execution failed")
+		}
+	}
+
+	availableTemplates := make(map[string]void)
+	logger.Debugf("Templates path: %s", nucleiTemplatePath)
+	tf, err := ioutil.ReadDir(nucleiTemplatePath)
+	if err != nil {
+		return errors.New("unable to read nuclei templates folder")
+	}
+
+	for _, f := range tf {
+		if f.IsDir() {
+			availableTemplates[f.Name()] = void{}
+		}
+	}
+	logger.Infof("Available templates: %+v", availableTemplates)
+
+	selectedTemplates := make(map[string]void)
+	inclusionList := strings.Split(opt.TemplateInclusionList, ",")
+	inclusionList = append(inclusionList, defaultInclusionList...)
+	// If inclusionList is not empty, use only inclued ones.
+	// If inclusionList is empty, user all available templates.
+	logger.Infof("Included templates: %+v", inclusionList)
+	if len(inclusionList) > 1 {
+		for _, v := range inclusionList {
+			if _, ok := availableTemplates[v]; ok {
+				selectedTemplates[v] = void{}
+			}
+		}
+	} else {
+		for k := range availableTemplates {
+			selectedTemplates[k] = void{}
+		}
+	}
+
+	// Remove exclued templates from selectedTemplates.
+	exclusionList := strings.Split(opt.TemplateExclusionList, ",")
+	exclusionList = append(exclusionList, defaultExclusionList...)
+	logger.Infof("Exclued templates: %+v", exclusionList)
+	for _, v := range exclusionList {
+		if _, ok := selectedTemplates[v]; ok {
+			delete(selectedTemplates, v)
+		}
+	}
+
+	logger.Infof("Selected templates: %+v", selectedTemplates)
+	// No templates selected. Return.
+	if len(selectedTemplates) < 1 {
+		return nil
+	}
+
+	// Build arguments.
+	nucleiArgs := []string{
+		"-target", target,
+		"-c", "20",
+		"-json",
+		"-silent",
+		"-H", userAgent,
+	}
+
+	// Include only selected severities.
+	if opt.Severities != "" {
+		logger.Infof("Selected severities: %s", opt.Severities)
+		severities := []string{"-severity", opt.Severities}
```

Comment:
Despite being an extra step, since the options are supplied in JSON, I would prefer to use an array of strings instead of a comma separated string for those, as we do in the ZAP check: https://github.com/adevinta/vulcan-checks/blob/master/cmd/vulcan-zap/manifest.toml#L8

This is not a strong opinion just an observation.

Reply:
Done.